options.js

```diff
@@ -1,139 +1,193 @@
+				var kanjiInfoObject = {};
+				var kanjiInfoLabelList = chrome.extension.getBackgroundPage().rcxDict.prototype.kanjiInfoLabelList;
+				var kanjiinfoarray = new Array(kanjiInfoLabelList.length/2);
+
+				// Setting Kanji values
+				for (i = 0; i*2 < kanjiInfoLabelList.length; i++) {
+					kanjiInfoObject[kanjiInfoLabelList[i * 2]] = document.getElementById(kanjiInfoLabelList[i * 2]).checked;
+					kanjiinfoarray[i] = kanjiInfoObject[kanjiInfoLabelList[i * 2]];
+				}
+
+				chrome.storage.sync.set({
+					// Saving General options
+					"popupcolor": popupcolor,
+					"highlight": highlight,
+					"textboxhl": textboxhl,
+					"onlyreading": onlyreading,
+					"minihelp": minihelp,
+					"disablekeys": disablekeys,
+					"kanjicomponents": kanjicomponents,
+					"kanjiInfo":kanjiInfoObject,
+					"popupLocation":popupLocation,
+					// Saving Copy to Clipboard settings
+					"lineEnding": lineEnding,
+					"copySeparator": copySeparator,
+					"maxClipCopyEntries": maxClipCopyEntries,
+					"popupDelay": popupDelay,
+					"showOnKey":showOnKey
+				});
+
+				chrome.extension.getBackgroundPage().rcxMain.config.css = popupcolor;
+				chrome.extension.getBackgroundPage().rcxMain.config.highlight = highlight;
+				chrome.extension.getBackgroundPage().rcxMain.config.textboxhl = textboxhl;
+				chrome.extension.getBackgroundPage().rcxMain.config.onlyreading = onlyreading;
+				chrome.extension.getBackgroundPage().rcxMain.config.minihelp = minihelp;
+				chrome.extension.getBackgroundPage().rcxMain.config.popupDelay = popupDelay;
+				chrome.extension.getBackgroundPage().rcxMain.config.disablekeys = disablekeys;
+				chrome.extension.getBackgroundPage().rcxMain.config.showOnKey = showOnKey;
+				chrome.extension.getBackgroundPage().rcxMain.config.kanjicomponents = kanjicomponents;
+				chrome.extension.getBackgroundPage().rcxMain.config.kanjiinfo = kanjiinfoarray;
+				chrome.extension.getBackgroundPage().rcxMain.config.lineEnding = lineEnding;
+				chrome.extension.getBackgroundPage().rcxMain.config.copySeparator = copySeparator;
+				chrome.extension.getBackgroundPage().rcxMain.config.maxClipCopyEntries = maxClipCopyEntries;
+				chrome.extension.getBackgroundPage().rcxMain.config.popupLocation = popupLocation;
+
+			}
+
+
+			function getPopUpDelay(){
+				// Todo: Why do we need a popup delay?
+				var popupDelay;
+				try {
+					popupDelay = parseInt(document.optform.popupDelay.value);
+					if (!isFinite(popupDelay)) {
+						throw Error('infinite');
+					}
+				} catch (err) {
+					popupDelay = 150;
+					popupDelay: "150";
+				}
+				return popupDelay;
+			}
+			window.onload = get_options;
+
+			/*function clicktab(tab) {
+				selectedtab = document.getElementById(tab);
+				// change format of all tabs to deselected
+				// change format of selected tab to selected
+				// hide all tab contents
+				// show selected tab contents
+			}*/
+
+
+			document.querySelector('#submit').addEventListener('click', save_options);
```

Comment:
this is also duplicated below, etc.

Probably all of this duplication would be fixed if you closed the for loop above correctly.

Reply:
FIXED THAT WEIRD STUFF!